### PR TITLE
Change the DataFusion explain plans to make it clearer in the predicate/filter

### LINKED ISSUE #2021: Change the DataFusion explain plans to make it clearer that the predicate / filter in the scan may be inexact / partially applied
I wonder if we could/should change the DataFusion explain plans to make it clearer that the predicate / filter in the scan may be inexact / partially applied

In other words, if `TableProvider:: supports_filter_pushdown` returns `TableProviderFilterPushDown::Inexact`

https://github.com/apache/arrow-datafusion/blob/d31c1579fdc2ad060bc46c4fcfef14cc7676da6b/datafusion/src/datasource/datasource.rs#L116-L123

https://github.com/apache/arrow-datafusion/blob/d31c1579fdc2ad060bc46c4fcfef14cc7676da6b/datafusion/src/datasource/datasource.rs#L54-L67

_Originally posted by @alamb in https://github.com/apache/arrow-datafusion/issues/2015#issuecomment-1068820110_

### PR DESCRIPTION
**Pull request by Ted-Jiang**
# Which issue does this PR close?

Closes #2021.

 # Rationale for this change
Thanks @alamb 's desgin to decouple between explain  and planNode, No need to change `tableScan` struct.
Now 
```
❯  explain select c1, c2 from test where  c2 = 0.000001;
+---------------+------------------------------------------------------------------------------------------------------------------------------------------------------+
| plan_type     | plan                                                                                                                                                 |
+---------------+------------------------------------------------------------------------------------------------------------------------------------------------------+
| logical_plan  | Projection: #test.c1, #test.c2                                                                                                                       |
|               |   Filter: #test.c2 = Float64(0.000001)                                                                                                               |
|               |     TableScan: test projection=Some([0, 1]), partial_filters=[#test.c2 = Float64(0.000001)]                                                          |
| physical_plan | ProjectionExec: expr=[c1@0 as c1, c2@1 as c2]                                                                                                        |
|               |   CoalesceBatchesExec: target_batch_size=4096                                                                                                        |
|               |     FilterExec: CAST(c2@1 AS Float64) = 0.000001                                                                                                     |
|               |       RepartitionExec: partitioning=RoundRobinBatch(16)                                                                                              |
|               |         CsvExec: files=[/Users/yangjiang/CLionProjects/github/arrow-datafusion/testing/data/csv/aggregate_test_100.csv], has_header=true, limit=None |
|               |                                                                                                                                                      |
+---------------+------------------------------------------------------------------------------------------------------------------------------------------------------+
2 rows in set. Query took 0.004 seconds.
``` 

# What changes are included in this PR?
plan tree node `TableScan`

# Are there any user-facing changes?

### CODE REVIEW

**Comment by yjshen** [comment]
We lost filters and projections in `ParquetExec`, lost projections for `CSV`, etc. What do you think if we include these missing parts for physical scan operators as well?

**Comment by Ted-Jiang** [comment]
> We lost filters and projections in `ParquetExec`, lost projections in `CSVExec`, etc. What do you think if we include these missing parts for physical scan operators as well?

Sure! may i create a sperate issue for these physican plan fmt change (avoid blend them together).

File: datafusion/src/optimizer/filter_push_down.rs [hunk 1]
```diff
@@ -1431,7 +1431,7 @@ mod tests {
         let plan = table_scan_with_pushdown_provider(TableProviderFilterPushDown::Exact)?;
 
         let expected = "\
-        TableScan: test projection=None, filters=[#a = Int64(1)]";
+        TableScan: test projection=None, full_filters=[#a = Int64(1)]";
```
Comment (alamb) [inline datafusion/src/optimizer/filter_push_down.rs:1434]: 👍

File: datafusion/src/logical_plan/plan.rs [hunk 2]
```diff
@@ -913,7 +916,39 @@ impl LogicalPlan {
                         )?;
 
                         if !filters.is_empty() {
-                            write!(f, ", filters={:?}", filters)?;
+                            let mut full_filter = vec![];
```
Comment (alamb) [inline datafusion/src/logical_plan/plan.rs:919]: this is quite nice

**Review by alamb — APPROVED** [review]

**Comment by alamb** [comment]
>  We lost filters and projections in ParquetExec, lost projections in CSVExec, etc. What do you think if we include these missing parts for physical scan operators as well?


I don't fully understand what you are suggesting here @yjshen  -- are you suggesting:
1. That this PR loses some information? 
2. A follow on PR to add more information to the `ParquetExec` would be helpful? 
3. Something else?

Thanks again @Ted-Jiang  -- this looks very helpful to avoid confusion

**Comment by yjshen** [comment]
> A follow on PR to add more information to the ParquetExec would be helpful?
Yes, Physical Table Scan Operators (ParquetExec, CsvExec ...) are losing information such as projection and predicate while being explained. We could be in a follow-up PR.

**Review by yjshen — APPROVED** [review]
LGTM. Thanks @Ted-Jiang! Let's clarify explain for physical scan operators in a follow-up PR.

**Pull request merged**